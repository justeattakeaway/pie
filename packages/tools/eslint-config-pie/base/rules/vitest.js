const vitest = require('eslint-plugin-vitest');

module.exports = {
    plugins: [
        'vitest',
    ],
    rules: {
        ...vitest.configs.recommended.rules,
    },
    globals: {
        ...vitest.environments.env.globals,
    },
};
