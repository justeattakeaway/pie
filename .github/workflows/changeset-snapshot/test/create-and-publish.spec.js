import { beforeEach, describe, expect, test, vi } from 'vitest';
const workflow = require('../create-and-publish');
const { Readable } = require('stream');

let context;
let execa;
let expectedBody;
let github;
let sampleOutput = '';

const mockExecaCommand = () => {
    const mockStdoutStream = new Readable({
        read: () => {
            this.push(sampleOutput);
            this.push(null); // End the stream
        },
    });

    const mockProcess = {
        stdout: mockStdoutStream,
        then: (resolve) => {
            setTimeout(() => resolve({
                stdout: sampleOutput,
            }), 0);
            return this;
        },
    };

    mockStdoutStream.pipe = vi.fn().mockImplementation((destination) => destination);

    return mockProcess;
  };

describe('create and publish workflow', () => {
    beforeEach(() => {
        sampleOutput = '';

        context = {
            actor: 'pie-contributor',
            issue: {
                number: 999,
            },
            repo: {
                owner: 'justeattakeaway',
                repo: 'pie',
            },
        };

        execa = {
            command: mockExecaCommand,
        };

        github = {
            rest: {
                issues: {
                    createComment: vi.fn().mockImplementation(({ body }) => {
                        expectedBody = body;
                    }),
                },
            },
        };
    });

    test('should create a github comment', async () => {
        // Act
        await workflow({ context, github }, execa);

        // Assert
        expect(github.rest.issues.createComment).toHaveBeenCalledTimes(1);
    });

    describe('if no packages were changed', () => {
        test('should have a message body starting with "No packages changed!"', async () => {
            // Arrange
            sampleOutput = '';

            // Act
            await workflow({ context, github }, execa);

            // Assert
            expect(expectedBody.startsWith('No changed packages found!')).toBe(true);
        });
    });

    describe('if other packages were changed', () => {
        test('should ignore monorepo/docs/storybook changes', async () => {
            // Arrange
            sampleOutput = `
                New tag: pie-docs@0.0.0-snapshot-release-20231220110000
                New tag: pie-monorepo@0.0.0-snapshot-release-20231220110000
                New tag: pie-storybook@0.0.0-snapshot-release-20231220110000
            `;

            // Act
            await workflow({ context, github }, execa);

            // Assert
            expect(expectedBody).toMatchSnapshot();
        });

        test('should ignore example app changes', async () => {
            // Arrange
            sampleOutput = `
                New tag: wc-vanilla@0.0.0-snapshot-release-20231220110000
                New tag: wc-next13@0.0.0-snapshot-release-20231220110000
            `;

            // Act
            await workflow({ context, github }, execa);

            // Assert
            expect(expectedBody).toMatchSnapshot();
        });
    });

    describe('if exactly one component was changed', () => {
        test('should not include the note about multiple packages', async () => {
            // Arrange
            sampleOutput = `
                New tag: @justeattakeaway/pie-button@0.0.0-snapshot-release-20231220110000
            `;

            // Act
            await workflow({ context, github }, execa);

            // Assert
            expect(expectedBody).toMatchSnapshot();
        });
    });

    describe('if more than one component was changed', () => {
        test('should include content related to multiple packages', async () => {
            // Arrange
            sampleOutput = `
                New tag: @justeattakeaway/pie-button@0.0.0-snapshot-release-20231220110000
                New tag: @justeattakeaway/pie-modal@0.0.0-snapshot-release-20231220110000
            `;

            // Act
            await workflow({ context, github }, execa);

            // Assert
            expect(expectedBody).toMatchSnapshot();
        });
    });
});
