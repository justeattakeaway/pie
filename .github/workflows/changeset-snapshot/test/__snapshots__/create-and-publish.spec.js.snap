// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`create and publish workflow > if exactly one component was changed > should not include the note about multiple packages 1`] = `
"@pie-contributor Your snapshot has been published to npm!

Test the snapshot by updating your \`package.json\` with the newly-published version:
\`\`\`sh
yarn up @justeattakeaway/pie-button@0.0.0-snapshot-release-20231220110000
\`\`\`"
`;

exports[`create and publish workflow > if more than one component was changed > should include content related to multiple packages 1`] = `
"@pie-contributor Your snapshots have been published to npm!

Test the snapshots by updating your \`package.json\` with the newly-published versions:
> [!NOTE]
> If you have more than one of these packages installed, we suggest using the new snapshots for all of them to help avoid version conflicts.

\`\`\`sh
yarn up @justeattakeaway/pie-button@0.0.0-snapshot-release-20231220110000 --mode=update-lockfile
\`\`\`
\`\`\`sh
yarn up @justeattakeaway/pie-modal@0.0.0-snapshot-release-20231220110000 --mode=update-lockfile
\`\`\`
Then finally:
\`\`\`sh
yarn install
\`\`\`"
`;

exports[`create and publish workflow > if other packages were changed > should ignore example app changes 1`] = `"No changed packages found! Please make sure you have added a changeset entry for the packages you would like to snapshot."`;

exports[`create and publish workflow > if other packages were changed > should ignore monorepo/docs/storybook changes 1`] = `"No changed packages found! Please make sure you have added a changeset entry for the packages you would like to snapshot."`;
